packages/vm/src/crypto.rs

```diff
@@ -174,4 +196,33 @@ mod tests {
             );
         }
     }
+
+    #[test]
+    fn test_ed25519_verify() {
+        // Explicit hash
+        let message_hash = Sha256::new().chain(MSG).finalize();
+
+        // Signing
+        let secret_key = ed25519::SigningKey::new(&mut OsRng);
+        let signature = secret_key.sign(&message_hash);
+
+        let public_key = ed25519::VerificationKey::from(&secret_key);
+
+        // Serialization. Types can be converted to raw byte arrays with From/Into
+        let signature_bytes: [u8; 64] = signature.into();
+        let public_key_bytes: [u8; 32] = public_key.into();
+
+        // Verification
+        assert!(ed25519_verify(&message_hash, &signature_bytes, &public_key_bytes).is_ok());
+
+        // Wrong message fails
+        let bad_message_hash = Sha256::new().chain([MSG, "\0"].concat()).finalize();
+        assert!(ed25519_verify(&bad_message_hash, &signature_bytes, &public_key_bytes).is_err());
+
+        // Other pubkey fails
+        let other_secret_key = ed25519::SigningKey::new(&mut OsRng);
+        let other_public_key = ed25519::VerificationKey::from(&other_secret_key);
+        let other_public_key_bytes: [u8; 32] = other_public_key.into();
+        assert!(ed25519_verify(&message_hash, &signature_bytes, &other_public_key_bytes).is_err());
```

Comment:
Here you find some external test vectors: https://github.com/cosmos/cosmjs/blob/v0.24.0-alpha.22/packages/crypto/src/libsodium.spec.ts#L278-L384